### PR TITLE
Editor: remove mask from text element output

### LINKED ISSUE #4721: `.mask` in text element causing undesired cropping
Many Web Stories have part of their text cropped at the bottom. See example: 

Per @merapi's  review, this is due to .mask `overflow:hidden` which should be removed from the text element. 
Per @choumx's review, .mask is also causing issues with text padding in some cases.

Suggestion: remove .mask unless necessary for something else, otherwise remove `overflow: hidden` regardless, per @merapi discussion with text-sets channel. 

Needs to be properly reviewed and QA'd to ensure no adverse negative effects.

### PR DESCRIPTION
**Pull request by embarks**
## Summary
- Per discussion [in the text sets channel](https://xwp.slack.com/archives/C01A6TD8HBJ/p1598445275005300) we should not apply `.mask`/`overflow: hidden` to all elements. This PR checks for the text type and removes the class which applied `overflow: hidden` to the text. Removing `.mask` from all elements

## Relevant Technical Choices

- removing `overflow: hidden` from all elements causes the video and image element crop to fail, so I **only** removed `.mask` from `text` type elements.

## To-do

- [x] percy screenshot tests
- [x] select a fancy font that bleeds over instead of basic `addTextElement`. Right now my tests are kind of lame... 
- [x] e2e test is more suitable for a Jest unit test in assets/src/edit-story/elements/text/test/output.js; less overhead. Could use a more generalized not.toHaveStyle('overflow: hidden') assertion instead of checking for the class name specifically.

I wasn't able to replicate the font cut-off from the ticket so steps to reproduce that might be useful...

## User-facing changes

- should fix the issue where certain text and glyphs would not overflow their containing box, resulting in undesired cropping e.g. 

Screen Shot 2020-11-03 at 4 02 07 PM
vs 
Screen Shot 2020-11-03 at 4 01 29 PM

## Testing Instructions
- create a new story
- add a text element to the page
- select "preview" 
- inspect the text element
- confirm its containing element no longer has the "mask" class:

### with mask 
Screen Shot 2020-11-05 at 3 13 17 PM

### without mask
Screen Shot 2020-11-05 at 3 08 06 PM

---

Fixes #4721

### CODE REVIEW

File: tests/e2e/specs/editor/text/noMask.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+ */
+import { percySnapshot } from '@percy/puppeteer';
+
+/**
+ * Internal dependencies
+ */
+import { createNewStory, previewStory, addTextElement } from '../../../utils';
+
+describe('Inserting text', () => {
+  it('should insert a text element with no mask', async () => {
```
Comment (swissspidy) [inline tests/e2e/specs/editor/text/noMask.js]: This particular test might be more suitable for a Jest unit test in `assets/src/edit-story/elements/text/test/output.js`; less overhead.

Could use a more generalized `not.toHaveStyle('overflow: hidden')` assertion instead of checking for the class name specifically.

File: tests/e2e/specs/editor/text/noMask.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    );
+
+    const textElementSelector = 'span[data-text=true]';
+    await page.waitForSelector(textElementSelector);
+    await expect(page).toMatchElement(textElementSelector, {
+      text: 'yes!',
+    });
+
+    await expect(page).toClick('li#library-tab-media');
+    await expect(page).toClick('div#wpbody');
```
Comment (swissspidy) [inline tests/e2e/specs/editor/text/noMask.js]: Is this to exit edit mode? Or what does it do?

File: tests/e2e/specs/editor/text/noMask.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+    const editorPage = page;
+    const previewPage = await previewStory(editorPage);
+
+    await percySnapshot(previewPage, 'Text glyphs are not cut off');
```
Comment (swissspidy) [inline tests/e2e/specs/editor/text/noMask.js]: Unfortunately the Percy snapshot isn't very useful at the moment...:

ceacef0cad045f6cec5e493c5c3256f17ed346aaca0aa84ad6f8551d34d108f5

Not sure why though. Looks like none of the AMP scripts have been loaded yet?

Might be unreliable to do a snapshot here :S

File: tests/e2e/specs/editor/text/noMask.js (hunk 2 above)
Comment (embarks) [reply tests/e2e/specs/editor/text/noMask.js]: yeah, that's what it's doing. I was having trouble with this. When I was calling `previewStory(editorPage)` without exiting edit mode it would just time out.

File: tests/e2e/specs/editor/text/noMask.js (hunk 3 above)
Comment (embarks) [reply tests/e2e/specs/editor/text/noMask.js]: lol, that's actually what I expected. Normally you'd see the `y` cut off on the bottom left. Maybe the text should be larger?

Comment (swissspidy) [reply tests/e2e/specs/editor/text/noMask.js]: But why is the background all white? I was expectin the actual story player with the arrow controls, the page in the center, etc. Just confusing if it's like this.

Comment (embarks) [reply tests/e2e/specs/editor/text/noMask.js]: ohh, perhaps I have to `await` something... is there a way for me to see the screenshots locally?

Comment (swissspidy) [reply tests/e2e/specs/editor/text/noMask.js]: Could try to `waitForSelector('amp-story.i-amphtml-layout')`.

As for running percy locally, you'd need the token from https://percy.io/google/web-stories-wp/settings and then run `PERCY_TOKEN=.... npm run test:e2e:percy`. That should work.

I had some issues in the past with getting percy / e2e tests to work with the story player on the frontend, so it might not be a trivial thing

Comment (swissspidy) [reply tests/e2e/specs/editor/text/noMask.js]: Related discussion: https://xwp.slack.com/archives/CCCAUAH6F/p1605022548428300

**Comment by embarks** [comment]
I removed the e2e tests per discussion. Replaced with checks on text and mask output.

**Review by carloskelly13 — APPROVED** [review]

File: assets/src/edit-story/elements/text/test/output.js [hunk 4]
```diff
@@ -221,7 +249,7 @@ describe('TextOutput', () => {
         box: { width: 1080, height: 1920, x: 50, y: 100, rotationAngle: 0 },
       };
 
-      await expect(<TextOutput {...props} />).toBeValidAMPStoryElement();
+      expect(<TextOutput {...props} />).toBeValidAMPStoryElement();
```
Comment (swissspidy) [inline assets/src/edit-story/elements/text/test/output.js]: Unrelated change? `toBeValidAMPStoryElement` is still async

Comment (embarks) [reply assets/src/edit-story/elements/text/test/output.js]: oops! yeah, before I rebased I was missing an eslint package and deleted these in a panic. ;)

**Comment by miina** [comment]
@embarks Looks like removing the mask from the text element is causing a regression: https://github.com/google/web-stories-wp/issues/5324

Can you confirm why is removing the mask necessary for text sets?

Edit: can see now that it's generally necessary for the letters to be all visible.
